examples/http/server.rb

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'rubygems'
+require 'bundler/setup'
+require 'sinatra/base'
+# Require otel-ruby
+require 'opentelemetry/sdk'
+
+SDK = OpenTelemetry::SDK
+OpenTelemetry.tracer_factory = SDK::Trace::TracerFactory.new
+
+exporter = SDK::Trace::Export::ConsoleSpanExporter.new
+processor = SDK::Trace::Export::SimpleSpanProcessor.new(exporter)
+OpenTelemetry.tracer_factory.add_span_processor(processor)
+
+# Rack middleware to extract span context, create child span, and add
+# attributes/events to the span
+class OpenTelemetryMiddleware
+  def initialize(app)
+    @app = app
+    @formatter = formatter = OpenTelemetry.tracer_factory.http_text_format
+    @tracer = OpenTelemetry.tracer_factory.tracer('sinatra', 'semver:1.0')
+  end
+
+  def call(env)
+    # Extract context from request headers
+    context = @formatter.extract(env) { |carrier, key| carrier[key] }
```

Comment:
@fbogsany @mwear I have the `context`, now what do I do with it? 😆 I didn't see any tests or examples that made this clear in how I can pass this context to the span.

Reply:
We talked about this briefly, but ultimately we want to use the extracted context to parent the span we're starting on the server. `in_span` doesn't appear to have a `with_parent_context` argument. Is that intentional @fbogsany? If not, then we should probably add options for `with_parent_context` and `with_parent`.

Reply:
@mwear I had a spike on this branch that added `with_parent` and `with_parent_context` to `in_span` and it seemed to work. If we want to go that way I'll add some tests and make sure it works as advertised.

Reply:
Not intentional. It should have the same signature as `start_span`.

Reply:
@mwear @fbogsany I've turned this into [a small PR](https://github.com/open-telemetry/opentelemetry-ruby/pull/144) so we don't clutter this one up any more than it is.

Reply:
@fbogsany I have removed the explicit block now.